### DISCUSSION TITLE
Prevent nodes from being removed

### DISCUSSION THREAD

**Discussion by rubenduiveman** [bcakmakoglu/vue-flow#275] [Q&A] [answered]
I have a use case where I want to prevent some nodes from being removed. I'd like to be able to "lock" a node by setting a property like `removable: false`, to prevent the remove button and keybinding to work.

**Comment by bcakmakoglu** [comment, marked as answer]
I see, that could be added at some point.
For now you can hook into the apply changes process and stop the deletion of nodes if necessary.

```vue
<script setup>
const { applyNodeChanges, onNodesChange } = useVueFlow()

onNodesChange((changes) => {
	const nextChanges = changes.filter((change) => {
		if (change.type === 'remove') {
			// block removing
		}
	})

	applyNodeChanges(nextChanges)
}) 
</script>

<template>
	<div style="height: 300px">
		<VueFlow :apply-default="false" />
	</div>
```

**Reply by rubenduiveman** [reply]
Excellent! The apply-default was what I was looking for! This works for me! Thanks!
